temoa/_internal/table_writer.py

```
@@ -96,425 +210,385 @@ def write_results(
-
-        data = []
-        scenario = (
-            self.config.scenario + f'-{iteration}'
-            if iteration is not None
-            else self.config.scenario
-        )
-        for ei in self.emission_register:
-            sector = self.tech_sectors[ei.t]
-            val = self.emission_register[ei]
+        if self.tech_sectors is None or self.emission_register is None:
+            raise RuntimeError('Dependencies missing (tech_sectors or emission_register)')
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)